### ISSUE TITLE
[BUG] Function call using ?* where context item is XPath 3.1 array gives syntax error err:XPST0003 

### ISSUE THREAD

**Issue by martin-honnen** [eXist-db/exist#3491]
When trying to execute code using a function call with the XPath/XQuery 3.1 `?` operator in the form of  `?*` eXist gives me a syntax error "err:XPST0003 expecting closing parenthesis ')', found '*'" while other XQuery 3.1 implementations (Saxon, BaseX) run the code fine.

Simplest test code I could come up with is

```xquery
([1,2,3], ['a', 'b', 'c'])!string-join(?*, '|')
```
gives an error 

> Cannot compile xquery: exerr:ERROR org.exist.xquery.XPathException: err:XPST0003 expecting closing parenthesis ')', found '*' 

**Describe the bug**
eXist gives a syntax error for a valid XQuery/XPath 3.1 expression.

**Expected behavior**
The result for the sample would be a sequence of two strings `1|2|3` and `a|b|c`, in general the `?*` should work as a lookup operator (https://www.w3.org/TR/xquery-31/#id-lookup) where the context item is a map or an array.

**To Reproduce**

Run 

```xquery
xquery version "3.1";

([1,2,3], ['a', 'b', 'c'])!string-join(?*, '|')
```
in eXide.

**Comment by joewiz** [comment]
@martin-honnen I believe this is a duplicate of https://github.com/eXist-db/exist/issues/3240. I'll paste your specific test code into that issue so that it's included in the resulting tests.

**Comment by line-o** [comment]
I think this is a separate problem whereas the question mark `?` is parsed as a parameter placeholder.
The error message hints at that

**Comment by line-o** [comment]
also related https://github.com/eXist-db/exist/issues/1655

**Comment by line-o** [comment]
Even after #3496 is merged this will still not work.
You need to add the context item explicitly, until the parser properly handles unary lookups, like so
```xquery
([1,2,3], ['a', 'b', 'c']) ! string-join(.?*, '|')
```

**Comment by line-o** [comment]
smallest test case for arrays `[1] ! ?*` and for maps `map{1:1} ! ?*`
